File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       int ptr1 = 0, ptr2 = nums.size()-1;
       while(ptr1!=nums.size()){
            int val = target - nums[ptr1];

            while(nums[ptr2] >= val)
            {
                if(nums[ptr2] == val)
                return{ptr1+1, ptr2+1};

                ptr2--;
            }

            ptr1++;

       }
       return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.go
func twoSum(nums []int, target int) []int {
    

    a := make(map[int]int)

    for i:=0; i<len(nums); i++{
        if _, ok := a[target - nums[i]]; ok{
            return []int{a[target - nums[i]], i}
        }

        a[nums[i]] = i
            
    }

    return []int{}
}
